components/provider_cldr/src/transform/dates.rs

```diff
@@ -115,6 +115,83 @@ impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
+            skeletons: {
+                let mut skeleton_tuples: Vec<SkeletonTupleV1> = Vec::new();
+
+                // The CLDR keys for available_formats can have duplicate skeletons with either
+                // an additional variant, or with multiple variants for different plurals.
+                for (skeleton_str, pattern_str) in other.available_formats.0.iter() {
+                    let mut unique_skeleton = String::new();
+                    let mut variant_text = String::new();
+
+                    // Split out the unique skeleton component, and the variant text.
+                    //   "MMMMW-count-two"
+                    //   unique_skeleton: "MMMMW"
+                    //   variant_text: "-count-two"
+                    let chars = skeleton_str.chars();
+                    let mut target = &mut unique_skeleton;
+                    for ch in chars {
+                        if ch == '-' {
+                            target = &mut variant_text;
+                        }
+                        target.push(ch);
+                    }
+
+                    let skeleton_fields_v1 =
+                        match SkeletonFieldsV1::try_from(unique_skeleton.as_str()) {
+                            Ok(s) => s,
+                            Err(err) => match err {
+                                // Ignore unimplemented fields for now.
+                                SkeletonFieldsV1Error::SymbolUnimplemented(_) => continue,
+                                SkeletonFieldsV1Error::SymbolUnknown(symbol) => panic!(
+                                    "Unknown symbol {:?} in skeleton {:?}",
+                                    symbol, unique_skeleton
+                                ),
+                                SkeletonFieldsV1Error::FieldTooLong => {
+                                    panic!("Field too long in skeleton {:?}", unique_skeleton)
+                                }
+                                SkeletonFieldsV1Error::SymbolInvalid(symbol) => panic!(
+                                    "Symbol invalid {:?} in skeleton {:?}",
+                                    symbol, unique_skeleton
+                                ),
+                            },
+                        };
+
+                    if !variant_text.is_empty() && variant_text != "-alt-variant" {
+                        unimplemented!(
+                            "This skeleton string is not yet supported: {:?}",
+                            skeleton_str
+                        );
+                    }
+
+                    let pattern_v1 = PatternV1::try_from(pattern_str as &str)
+                        .expect("Unable to parse a pattern");
+
+                    skeleton_tuples.push(SkeletonTupleV1(skeleton_fields_v1, pattern_v1));
+                }
+
+                // Sort the skeletons deterministically by their canonical sort order.
+                skeleton_tuples.sort_by(|a, b| a.0.compare_canonical_order(&b.0));
```

Comment:
If you use LiteMap here, the sorting will be done automatically.

Reply:
This required some trait implementations, but LiteMap is now handling the sorting.

Reply:
This is: e12a4bb6761a4f7355791f292bc0711f89df3eea